packages/webapp/pages/onboard.tsx

```diff
@@ -0,0 +1,176 @@
+import { useRouter } from 'next/router';
+import { useAnalyticsContext } from '@dailydotdev/shared/src/contexts/AnalyticsContext';
+import {
+  AnalyticsEvent,
+  TargetType,
+} from '@dailydotdev/shared/src/lib/analytics';
+import { OnboardingStep } from '@dailydotdev/shared/src/components/onboarding/common';
+import { OnboardingMode } from '@dailydotdev/shared/src/graphql/feed';
+import { useAuthContext } from '@dailydotdev/shared/src/contexts/AuthContext';
+
+const versionToTitle: Record<OnboardingFilteringTitle, string> = {
+  [OnboardingFilteringTitle.Control]: 'Choose topics to follow',
+  [OnboardingFilteringTitle.V1]: 'What topic best describes you?',
+  [OnboardingFilteringTitle.V2]: 'Which topics resonate with you the most?',
+  [OnboardingFilteringTitle.V3]: `Pick the topics you'd love to dive into`,
+  [OnboardingFilteringTitle.V4]: 'Choose the topics you’re passionate about',
+};
+
+const Title = classed('h2', 'font-bold typo-title2');
+
+export function OnboardPage(): ReactElement {
+  const router = useRouter();
+  const { user, isAuthReady } = useAuthContext();
+  const [isFiltering, setIsFiltering] = useState(false);
+  const [isAuthenticating, setIsAuthenticating] = useState(false);
+  const { onShouldUpdateFilters } = useOnboardingContext();
+  const {
+    onboardingFilteringTitle,
+    onboardingMinimumTopics,
+    onboardingV2,
+    isFeaturesLoaded,
+  } = useFeaturesContext();
+  const { trackEvent } = useAnalyticsContext();
+
+  const onClickNext = () => {
+    const screen = isFiltering ? OnboardingStep.Topics : OnboardingStep.Intro;
+
+    trackEvent({
+      event_name: AnalyticsEvent.ClickOnboardingNext,
+      extra: JSON.stringify({ screen_value: screen }),
+    });
+
+    if (!isFiltering) return setIsFiltering(true);
+
+    return setIsAuthenticating(true);
+  };
+
+  const formRef = useRef<HTMLFormElement>();
+  const title = versionToTitle[onboardingFilteringTitle];
+  const percentage = isAuthenticating ? 100 : 50;
+  const content = isAuthenticating
+    ? 'Once you sign up, your personal feed\nwill be ready to explore.'
+    : `Pick a few subjects that interest you.\nYou can always change these later.`;
+
+  const onSuccessfulTransaction = () => {
+    onShouldUpdateFilters(true);
+    router.push('/');
+  };
+
+  const isPageReady = isFeaturesLoaded && isAuthReady;
+
+  useEffect(() => {
+    if (!isPageReady) return;
+
+    if (user || onboardingV2 === OnboardingV2.Control) {
+      router.push('/');
+      return;
+    }
+
+    trackEvent({
+      event_name: AnalyticsEvent.Impression,
+      target_type: TargetType.MyFeedModal,
+      target_id: onboardingV2,
+      extra: JSON.stringify({
+        origin: OnboardingMode.Wall,
+        steps: [OnboardingStep.Topics],
+        mandating_categories: onboardingMinimumTopics,
+      }),
+    });
+  }, [
+    trackEvent,
+    isPageReady,
+    onboardingV2,
+    onboardingMinimumTopics,
+    router,
+    user,
+  ]);
```

Comment:
@idoshamun tracking for impression. This also makes sure it doesn't send an `impression` when the page loads and the parameters are not yet considered.

Reply:
Can you please make sure it doesn't get triggered twice? The rest looks great!